src/emitterutils.cpp

```diff
@@ -157,7 +157,7 @@ bool IsValidPlainScalar(const std::string& str, FlowType::value flowType,
   }
 
   // check against null
-  if (str == "null") {
+  if (str == "~" || str == "null" || str == "Null" || str == "NULL") {
```

Comment:
Can you extract these two places into a separate function? Perhaps put it in null.h / null.cpp.